src/context-builder.ts

```diff
@@ -40,10 +42,16 @@ export class ContextBuilder {
     return this;
   }
 
+  pushSubcontext(...contexts: ReadonlyContext[]) {
+    this.subcontexts.push(...contexts);
+    return this;
+  }
+
```

Comment:
Before merging this I want experiment with inheritance of contexts to see if I can create separate contexts for each of the middlewares we provide. All of them would conform to a shared interface, so that some methods are easily callable without type casting.